hack/update-bazel.sh

```diff
@@ -26,13 +26,16 @@ kube::util::ensure-gnu-sed
 # TODO(spxtr): Remove this line once Bazel is the only way to build.
 rm -f "${KUBE_ROOT}/pkg/generated/openapi/zz_generated.openapi.go"
 
-# The git commit sha1s here should match the values in $KUBE_ROOT/WORKSPACE.
-kube::util::go_install_from_commit \
-    github.com/kubernetes/repo-infra/kazel \
-    97099dccc8807e9159dc28f374a8f0602cab07e1
-kube::util::go_install_from_commit \
-    github.com/bazelbuild/bazel-gazelle/cmd/gazelle \
-    578e73e57d6a4054ef933db1553405c9284322c7
+# Install tools we need, but only from vendor/...
+go install ./vendor/github.com/bazelbuild/bazel-gazelle/cmd/gazelle
+if ! which gazelle >/dev/null 2>&1; then
```

Comment:
this seems pretty fragile - it's can be possible that `$GOPATH/bin` is not in your PATH, but there might already be a gazelle installed in `$PATH` anyway, which will cause this to pass unexpectedly.

Maybe instead set `GOBIN="${KUBE_OUTPUT_BINPATH}"` (i.e. `$KUBE_ROOT/_output/bin`( before running `go install`?

this is what I did for https://github.com/kubernetes/test-infra/pull/7429 and https://github.com/kubernetes/repo-infra/pull/65 and it seems to be working well.

Reply:
Not a bad idea.  I forgot we have that.